src/App.tsx

```diff
@@ -1,14 +1,50 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [preparedTodos, setPreparedTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLoadingFailed, setIsLoadingFailed] = useState(false);
+
+  const getTodosFromServer = async () => {
+    setIsLoading(true);
+    try {
+      const todosFromServer = await getTodos();
+
+      setIsLoading(false);
+      setTodos(todosFromServer);
+      setPreparedTodos(todosFromServer);
+    } catch (error: unknown) {
+      setIsLoadingFailed(true);
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const selectTodo = (todoId: number) => {
+    setSelectedTodoId(todoId);
+  };
+
+  const closeModal = () => {
+    setSelectedTodoId(0);
+  };
+
+  const selectedTodo = todos.find(todo => todo.id === selectedTodoId) as Todo;
```

Comment:
why do you need type conversion here?

Reply:
I added it because the result of the find method could be undefined and I am passing selectedTodo to the TodoModal component as a prop. It gives me an error when I pass it .
 If todo is listed in the list we know that it exists this is why we are sure that we will find it  so I decided to go with type conversion.

Reply:
@nurdanokcu I guess this should work if you make it empty array if result of `find` is undefined